tests/linearGaussian_npse_test.py

```diff
@@ -1,300 +1,281 @@
-        prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov)
-        gt_posterior = true_posterior_linear_gaussian_mvn_prior(
-            x_o, likelihood_shift, likelihood_cov, prior_mean, prior_cov
-        )
-        target_samples = gt_posterior.sample((num_samples,))
-    else:
-        prior = utils.BoxUniform(-2.0 * ones(num_dim), 2.0 * ones(num_dim))
-        target_samples = samples_true_posterior_linear_gaussian_uniform_prior(
-            x_o,
-            likelihood_shift,
-            likelihood_cov,
-            prior=prior,
-            num_samples=num_samples,
-        )
-
-    inference = NPSE(prior, sde_type=sde_type, show_progress_bars=True)
 
-    theta = prior.sample((num_simulations,))
-    x = linear_gaussian(theta, likelihood_shift, likelihood_cov)
-
-    score_estimator = inference.append_simulations(theta, x).train(
-        training_batch_size=100
+sampling_test_cases_1_trial = [
+    pytest.param(
+        NpseSamplingTestCase(iid, sampling, 1), id=f"{iid}-{sampling}-trials_1"
     )
-    # amortize the training when testing sample_with.
-    for method in sample_with:
-        posterior = inference.build_posterior(score_estimator, sample_with=method)
-        posterior.set_default_x(x_o)
-        samples = posterior.sample((num_samples,))
-
-        # Compute the c2st and assert it is near chance level of 0.5.
-        check_c2st(
-            samples,
-            target_samples,
-            alg=f"npse-{sde_type or 'vp'}-{prior_str}-{num_dim}D-{method}",
-        )
-
-    # Checks for log_prob()
-    if prior_str == "gaussian":
-        # For the Gaussian prior, we compute the KLd between ground truth and
-        # posterior.
-        dkl = get_dkl_gaussian_prior(
-            posterior,
-            x_o[0],
-            likelihood_shift,
-            likelihood_cov,
-            prior_mean,
-            prior_cov,
-        )
-
-        max_dkl = 0.15
-
-        assert dkl < max_dkl, (
-            f"D-KL={dkl} is more than 2 stds above the average performance."
-        )
-
-
-def test_c2st_npse_on_linearGaussian_different_dims():
-    """Test NPE on linear Gaussian with different theta and x dimensionality."""
-
-    theta_dim = 3
-    x_dim = 2
-    discard_dims = theta_dim - x_dim
-
-    x_o = zeros(1, x_dim)
-    num_samples = 1000
-    num_simulations = 2000
-
-    # likelihood_mean will be likelihood_shift+theta
-    likelihood_shift = -1.0 * ones(x_dim)
-    likelihood_cov = 0.3 * eye(x_dim)
-
-    prior_mean = zeros(theta_dim)
-    prior_cov = eye(theta_dim)
-    prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov)
-    target_samples = samples_true_posterior_linear_gaussian_mvn_prior_different_dims(
-        x_o,
-        likelihood_shift,
-        likelihood_cov,
-        prior_mean,
-        prior_cov,
-        num_discarded_dims=discard_dims,
-        num_samples=num_samples,
+    for iid, sampling in product(IID_METHODS, SAMPLING_METHODS)
+]
```

Comment:
I think we can drop those as the iid methods only take effect with `num_trials>1`